### PR TITLE
Richlist command and webapi

### PR DESCRIPTION
**Pull request by zhiyuan2007**
uri: /richlist
args: n(int) & include-distribution(bool)
return values: [{address: ..., coins: ..., locked: bool}]
example:
http://127.0.0.1:6420/richlist?n=1&include-distribution=true
```
[
    {
        "address": "zMDywYdGEDtTSvWnCyc3qsYHWwj9ogws74",
        "coins": "1000000.000000",
        "locked": true
    }
]
```
also cli commands:
usage:
cli richlist ${topn}  ${include-distribution}
example
```
./cli richlist 1  true
{
    "richlist": [
        {
            "address": "zMDywYdGEDtTSvWnCyc3qsYHWwj9ogws74",
            "coins": "1000000.000000",
            "locked": true
        }
    ]
}
```

### CODE REVIEW

File: src/daemon/gateway.go [hunk 1]
```diff
@@ -657,3 +658,27 @@ func (gw *Gateway) GetBuildInfo() visor.BuildInfo {
...
+	if err != nil {
+		return topnAccount, err
+	}
+
+	allAccounts, err := rbOuts.AggregateUnspentOutputs()
+	if err != nil {
+		return topnAccount, err
+	}
+
+	distributionMap := visor.GetLockedDistributiomAddressMap()
```
Comment (gz-c) [inline src/daemon/gateway.go]: typo: `GetLockedDistributionAddressMap`

File: src/daemon/gateway.go [hunk 2]
```diff
@@ -657,3 +658,27 @@ func (gw *Gateway) GetBuildInfo() visor.BuildInfo {
 	})
 	return bi
 }
+
+// GetRichlist returns rich list as desc order.
+func (gw *Gateway) GetRichlist(topn int, includeDistribution bool) ([]uxotutil.AccountJSON, error) {
+	var topnAccount []uxotutil.AccountJSON
```
Comment (gz-c) [inline src/daemon/gateway.go]: `topnAccounts`

File: src/util/uxotutil/uxotutil.go [hunk 3]
```diff
@@ -0,0 +1,118 @@
+/*************************************************************************
```
Comment (gz-c) [inline src/util/uxotutil/uxotutil.go]: no need for this

File: src/util/uxotutil/uxotutil.go [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+   > Mail: zhiyuan_06@126.com
+   > Created Time: 一 10/16 23:09:41 2017
+************************************************************************/
+package uxotutil
+
+import (
+	"sort"
+	"strings"
+
+	"github.com/spaco/spo/src/util/droplet"
```
Comment (gz-c) [inline src/util/uxotutil/uxotutil.go]: wrong import

File: src/util/uxotutil/uxotutil.go [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+}
+
+func (am AccountMgr) Swap(i, j int) {
+	am.Accounts[i], am.Accounts[j] = am.Accounts[j], am.Accounts[i]
+}
+
+//Sort sort coin owner desc order
+func (am *AccountMgr) Sort() {
+	sort.Sort(am)
+	am.IsSorted = true
```
Comment (gz-c) [inline src/util/uxotutil/uxotutil.go]: is `IsSorted` necessary?

File: src/daemon/gateway.go [hunk 6]
```diff
@@ -657,3 +658,27 @@ func (gw *Gateway) GetBuildInfo() visor.BuildInfo {
...
+	rbOuts, err := gw.GetUnspentOutputs(FbyAddressesNotIncluded([]string{}))
+	if err != nil {
+		return topnAccount, err
+	}
+
+	allAccounts, err := rbOuts.AggregateUnspentOutputs()
+	if err != nil {
+		return topnAccount, err
```
Comment (gz-c) [inline src/daemon/gateway.go]: `return nil, err`

File: src/api/webrpc/webrpc_test.go [hunk 7]
```diff
@@ -106,6 +107,28 @@ func (fg fakeGateway) GetAddrUxOuts(addr cipher.Address) ([]*historydb.UxOutJSON
 	return nil, nil
 }
 
+func (fg fakeGateway) GetRichlist(topn int, includeDistribution bool) ([]uxotutil.AccountJSON, error) {
+	var topnAccount []uxotutil.AccountJSON
+	rbOuts, err := fg.GetUnspentOutputs(daemon.FbyAddressesNotIncluded([]string{}))
+	if err != nil {
+		return topnAccount, err
```
Comment (gz-c) [inline src/api/webrpc/webrpc_test.go]: `return nil, err`

File: src/api/webrpc/richlist.go [hunk 8]
```diff
@@ -0,0 +1,28 @@
+package webrpc
```
Comment (gz-c) [inline src/api/webrpc/richlist.go]: Do not add any rpc commands. RPC interface will be removed.

File: src/util/uxotutil/uxotutil.go [hunk 9]
```diff
@@ -0,0 +1,118 @@
...
+
+import (
+	"sort"
+	"strings"
+
+	"github.com/spaco/spo/src/util/droplet"
+)
+
+//AccountMgr manager all unspent outputs
+type AccountMgr struct {
```
Comment (gz-c) [inline src/util/uxotutil/uxotutil.go]: Is this new package necessary? Can put it in daemon or visor or gui package?

Comment (zhiyuan2007) [reply src/util/uxotutil/uxotutil.go]: move to visor

File: cmd/cli/README.md [hunk 10]
```diff
@@ -96,6 +96,7 @@ COMMANDS:
      walletOutputs         Display outputs of specific wallet
      addressBalance        Check the balance of specific addresses
      addressOutputs        Display outputs of specific addresses
+     richlist              Display rich list as desc order
```
Comment (gz-c) [inline cmd/cli/README.md]: Remove

**Comment by gz-c** [comment]
just needs docs

**Pull request merged**